flex/engines/graph_db/grin/predefine.h

```diff
@@ -0,0 +1,282 @@
+#define GRIN_TRAIT_SELECT_PARTITION_FOR_EDGE_LIST
+#define GRIN_TRAIT_SELECT_MASTER_NEIGHBOR_FOR_ADJACENT_LIST
+#define GRIN_TRAIT_SELECT_NEIGHBOR_PARTITION_FOR_ADJACENT_LIST
+*/
+
+// Property
+#define GRIN_ENABLE_ROW
+#define GRIN_TRAIT_CONST_VALUE_PTR
+#define GRIN_WITH_VERTEX_PROPERTY
+#define GRIN_WITH_VERTEX_PROPERTY_NAME
+#define GRIN_WITH_VERTEX_TYPE_NAME
+#define GRIN_TRAIT_NATURAL_ID_FOR_VERTEX_TYPE
+#define GRIN_ENABLE_VERTEX_PRIMARY_KEYS
+#define GRIN_TRAIT_NATURAL_ID_FOR_VERTEX_PROPERTY
+#define GRIN_WITH_EDGE_PROPERTY
+// #define GRIN_WITH_EDGE_PROPERTY_NAME
+#define GRIN_WITH_EDGE_TYPE_NAME
+#define GRIN_TRAIT_NATURAL_ID_FOR_EDGE_TYPE
+// #define GRIN_ENABLE_EDGE_PRIMARY_KEYS
+// #define GRIN_TRAIT_NATURAL_ID_FOR_EDGE_PROPERTY
+// #define GRIN_TRAIT_SPECIFIC_VEV_RELATION
+// #define GRIN_ASSUME_MASTER_ONLY_PARTITION_FOR_VERTEX_PROPERTY
+// #define GRIN_ASSUME_REPLICATE_MASTER_MIRROR_PARTITION_FOR_VERTEX_PROPERTY
+// #define GRIN_ASSUME_SPLIT_MASTER_MIRROR_PARTITION_FOR_VERTEX_PROPERTY
+// #define GRIN_ASSUME_MASTER_ONLY_PARTITION_FOR_EDGE_PROPERTY
+// #define GRIN_ASSUME_REPLICATE_MASTER_MIRROR_PARTITION_FOR_EDGE_PROPERTY
+// #define GRIN_ASSUME_SPLIT_MASTER_MIRROR_PARTITION_FOR_EDGE_PROPERTY
+//  Index
+#define GRIN_WITH_VERTEX_LABEL
+#define GRIN_WITH_EDGE_LABEL
+/// #define GRIN_ASSUME_ALL_VERTEX_LIST_SORTED
+#define GRIN_ENABLE_VERTEX_INTERNAL_ID_INDEX
+#define GRIN_ENABLE_VERTEX_PK_INDEX
+// #define GRIN_ENABLE_EDGE_PK_INDEX
+
+/* 3. Define the handles using typedef */
+typedef void* GRIN_GRAPH;
+typedef void* GRIN_VERTEX;
+typedef void* GRIN_EDGE;
+
+#ifdef GRIN_WITH_VERTEX_DATA
+typedef void* GRIN_VERTEX_DATA;
+#endif
+
+#ifdef GRIN_WITH_VERTEX_PROPERTY
+typedef unsigned GRIN_VERTEX_TYPE;
+typedef void* GRIN_VERTEX_TYPE_LIST;
+typedef void* GRIN_VERTEX_PROPERTY;
+typedef void* GRIN_VERTEX_PROPERTY_LIST;
+#endif
+
+#ifdef GRIN_ENABLE_VERTEX_LIST
+typedef struct GRIN_VERTEX_LIST {
+  size_t vertex_num;
+  GRIN_VERTEX_TYPE label;
+} GRIN_VERTEX_LIST;
+#endif
+
+#ifdef GRIN_ENABLE_VERTEX_LIST_ITERATOR
+typedef void* GRIN_VERTEX_LIST_ITERATOR;
+#endif
+
+#ifdef GRIN_ENABLE_ADJACENT_LIST
+typedef void* GRIN_ADJACENT_LIST;
+#endif
+
+#ifdef GRIN_ENABLE_ADJACENT_LIST_ITERATOR
+typedef void* GRIN_ADJACENT_LIST_ITERATOR;
+#endif
+
+#ifdef GRIN_WITH_EDGE_DATA
+typedef void* GRIN_EDGE_DATA;
+#endif
+
+#ifdef GRIN_ENABLE_EDGE_LIST
+typedef void* GRIN_EDGE_LIST;
+#endif
+
+#ifdef GRIN_ENABLE_EDGE_LIST_ITERATOR
+typedef void* GRIN_EDGE_LIST_ITERATOR;
+#endif
+
+/**
+#ifdef GRIN_ENABLE_GRAPH_PARTITION
+typedef void* GRIN_PARTITIONED_GRAPH;
+typedef void* GRIN_PARTITION;
+typedef void* GRIN_PARTITION_LIST;
+#endif
+
+#ifdef GRIN_TRAIT_NATURAL_ID_FOR_PARTITION
+typedef unsigned GRIN_PARTITION_ID;
+#endif
+*/
+
+#ifdef GRIN_ENABLE_VERTEX_REF
+typedef void* GRIN_VERTEX_REF;
+#endif
+
+#ifdef GRIN_ENABLE_EDGE_REF
+typedef void* GRIN_EDGE_REF;
+#endif
+
+#ifdef GRIN_TRAIT_NATURAL_ID_FOR_VERTEX_TYPE
+typedef GRIN_VERTEX_TYPE GRIN_VERTEX_TYPE_ID;
```

Comment:
although GRIN_VERTEX_TYPE is defined as unsigned, but here keeping it as typedef unsigned GRIN_VERTEX_TYPE_ID is much clearer